### ISSUE TITLE
Finalize registration

### ISSUE THREAD

**Issue by kidandcat** [SAP/gigya-flutter-plugin#13]
Hi, we have required fields in our user schema, and so when users use social login, some fields are missing.

If I'm not wrong, we should call finalizeRegistration with the missing data:

https://help.sap.com/viewer/8b8d6fffe113457094a17701f63e3d6a/GIGYA/en-US/228cd8bc68dc477094b3e0e9fe108e23.html

Pointed in gigya's docs:

> Account pending registration (error code 206001) - returned when you call a method that performs social login, such as this method, and the registration process has not been finalized, or the schema defines fields as required and one or more of these fields are missing from the user Profile or custom data fields. The expected next step is: if the schema defines fields that are required and one or more of these fields are missing from the user profile or data, call accounts.setAccountInfo. If the registration process has not been finalized, call accounts.finalizeRegistration with the regToken.

How should we handle this situation with Flutter SDK? We tried to use `setAccount` to set missing fields, but we can't call `setAccount` if the user is not logged in. (and he can't log in if his registration process has not finished)

UPDATE:

I've advanced in the flux, which is similar to conflictingAccounts.

Now I need to update the account using the resolver:

                    resolver.setAccount({
                      "profile": json.encode({
                        "birthMonth": '5',
                        "birthYear": '5',
                      }),
                    });

I'm just getting this error:

```
E/flutter (22965): [ERROR:flutter/lib/ui/ui_dart_state.cc(199)] Unhandled Exception: type 'ArgumentError' is not a subtype of type 'PlatformException'
E/flutter (22965): #0      PendingRegistrationResolver.setAccount.<anonymous closure> (package:gigya_flutter_plugin/interruption/interruption_resolver.dart:72:48)
E/flutter (22965): #1      _rootRunUnary (dart:async/zone.dart:1362:47)
E/flutter (22965): #2      _CustomZone.runUnary (dart:async/zone.dart:1265:19)
E/flutter (22965): #3      _FutureListener.handleError (dart:async/future_impl.dart:172:20)
E/flutter (22965): #4      Future._propagateToListeners.handleError (dart:async/future_impl.dart:719:47)
E/flutter (22965): #5      Future._propagateToListeners (dart:async/future_impl.dart:740:24)
E/flutter (22965): #6      Future._completeError (dart:async/future_impl.dart:550:5)
E/flutter (22965): #7      _completeOnAsyncError (dart:async-patch/async_patch.dart:274:13)
E/flutter (22965): #8      MethodChannel.invokeMapMethod (package:flutter/src/services/platform_channel.dart)
E/flutter (22965): <asynchronous suspension>
```

- First, could you documentate how to use that method? 

- Second, there is an error trying to parse an exception as you can see above.

**Comment by tal-mi** [comment]
Hi,

The resolver pattern is the way to go here.
Here is a working code snippet:
```
if (error.getInterruption() == Interruption.pendingRegistration) {
        debugPrint('Pending registration interruption');
        PendingRegistrationResolver resolver =
            GigyaSdk.instance.resolverFactory.getResolver(error);
        resolver.setAccount({
          'profile': json.encode({
              'birthMonth': '5',
              'birthYear': '5',
            }),
        }).then((res) {
          final Account account = Account.fromJson(res);

          setState(() {
            _inProgress = false;
            _requestResult =
            'Login success:\n\n ${account.uid}';
          });
          Navigator.of(context).pop();
        });
      }
```

Also:
Make sure that all the profile required fields are set to "clientModify" on your site schema.

**Comment by tal-mi** [comment]
Hi,

Is this issue still persist?

Thanks,

**Comment by kidandcat** [comment]
We couldn't check it yet, I'll update as soon as we test it.

**Comment by kidandcat** [comment]
Fixed updating to 0.1.1
Thanks!
